### PR TITLE
feat: add built-in function to generate a UUID string

### LINKED ISSUE #796: Add built-in function to generate an UUID
**Is your feature request related to a problem? Please describe.**
We use UUID's a lot in requests and match with them in replies, so being able to generate a UUID from within FEEL itself would be a huge benefit.

Alternatives are generating one with a Service Worker, or even requesting one from an online source with the REST connector, but these look like aweful alternatives.

As requested here: https://forum.camunda.io/t/built-in-function-for-uuid/35018/14

**Describe the solution you'd like**

Being able to do `=UUID()` to create an input variable with an UUID.

```
// function signature
uuid(): String

// usage
uuid()    // -> "7793aab1-d761-4d38-916b-b7270e309894"
uuid()    // -> "df09d5c2-e9f0-432d-9027-aaa5db56b8b5"
```

**Related issues**

* https://github.com/camunda/zeebe/issues/10987

### PR DESCRIPTION
**Pull request by skayliu**
## Description

Add the following built-in function `uuid()`.

```
// function signature
uuid(): String

// usage
uuid()    // -> "7793aab1-d761-4d38-916b-b7270e309894"
uuid()    // -> "df09d5c2-e9f0-432d-9027-aaa5db56b8b5"
```
## Related issues

closes #796

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -90,6 +91,12 @@
       <artifactId>slf4j-api</artifactId>
     </dependency>
 
+    <dependency>
+      <groupId>com.fasterxml.uuid</groupId>
+      <artifactId>java-uuid-generator</artifactId>
+      <version>${version.uuid}</version>
+    </dependency>
```
Comment (saig0) [inline pom.xml:98]: :question: Why do you want to use this library?

Java has a built-in `UUID` class. You could generate UUID by using `UUID.randomUUID()`.

**Review by saig0 — COMMENTED** [review]
@skayliu thank you for your contribution. :tada: 

The changes look very good. :+1: Before accepting it, I have a question about the new dependency. Please have a look.

File: pom.xml (hunk 1 above)
Comment (skayliu) [reply pom.xml:98]: if you insist on the Java build-in `UUID.randomUUID()`, i can change to it.

Comment (saig0) [reply pom.xml:98]: @skayliu very interesting. :+1: I see that this library is also used by Camunda 7. :bulb: That is a good indication for using it here too.

File: src/main/scala/org/camunda/feel/impl/builtin/StringBuiltinFunctions.scala [hunk 2]
```diff
@@ -262,4 +264,13 @@ object StringBuiltinFunctions {
         ValString(string.trim)
       }
     )
+
+  private def uuidFunction =
+    builtinFunction(
+      params = List(),
+      invoke = { case List() =>
+        val uuid  = Generators.timeBasedGenerator.generate()
```
Comment (saig0) [inline src/main/scala/org/camunda/feel/impl/builtin/StringBuiltinFunctions.scala:272]: :wrench: In Camunda 7, we create only one instance of the generator, by including the ethernet address ([ref](https://github.com/camunda/camunda-bpm-platform/blob/master/engine/src/main/java/org/camunda/bpm/engine/impl/persistence/StrongUuidGenerator.java#L44C32-L44C94)). We could do the same and define the generate as lazy val field.

```
private lazy val generator = Generators.timeBasedGenerator(EthernetAddress.fromInterface())

// ...
invoke = { case List() => ValString(generator.generate.toString()) }
```

**Review by saig0 — COMMENTED** [review]
@skayliu I have one remaining minor comment about the generator usage. Please have a look if you agree. :cookie:

File: src/main/scala/org/camunda/feel/impl/builtin/StringBuiltinFunctions.scala (hunk 2 above)
Comment (skayliu) [reply src/main/scala/org/camunda/feel/impl/builtin/StringBuiltinFunctions.scala]: @saig0, It's done.

**Review by saig0 — APPROVED** [review]
@skayliu LGTM. :tada:

**Pull request merged**